Move the workload e2e tests to it's own package

wanghaoran1988 opened the pull request:
**What this PR does / why we need it**:

Move the workload e2e tests to it's own package

**Release note**:
```
None
```

wanghaoran1988 committed:
Move the workload e2e tests to it's own package

wanghaoran1988 commented:
/cc @kargakis @janetkuo @enisoc

kargakis commented:
@kubernetes/sig-apps-pr-reviews

kargakis commented:
@k8s-bot ok to test

kargakis commented:
Part of https://github.com/kubernetes/kubernetes/issues/7666

kargakis commented:
Alongside with this move we need to create a new workload-specific package where all the utilities will live. Currently we have utilities in two different places:
https://github.com/kubernetes/kubernetes/tree/master/test/utils
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go
I'd vote for a `test/utils/workloads` package.

wanghaoran1988 commented:
@kargakis Good points, will update tomorrow :)

wanghaoran1988 commented:
@k8s-bot pull-kubernetes-bazel test this

wanghaoran1988 commented:
@kargakis @ixdy Would you mind tell me why the bazel build failed, and I already run hack/update-bazel.sh .

wanghaoran1988 commented:
> I'd vote for a test/utils/workloads package.

@kargakis I'd prefer move them to test/e2e/framework, so that other components can also use them conveniently like how we do currently, let me know you have any other concerns.

kargakis commented:
framework is already bloated; workload-specific utilities should be in their own package. OWNERS files are not valuable when the bot has to pick somebody out of the whole organization.

fabianofranz commented:
@k8s-bot pull-kubernetes-unit test this
@k8s-bot pull-kubernetes-bazel test this

wanghaoran1988 commented:
@kargakis I am trying to work as you suggested, but it's hard to split the workloads func from the framework/util.go, there are some basic/common func in framework package that workload related funcs will use, after we split the workload related func to test/utils/workloads, this new package have to depend on the framework package, and some other resource utils under framework package need use the workloads func, this will introduce the cycle import in golang, it's really difficult to split clearly now, so to keep this pr simple, how about  address two problems in this pr:
1. E2e tests move to it's own package
2. Split every workload related resource utils have it's own util go file, deployment.go rc.go statefu_set.go .etc. this will make it a little easier to maintain currently.

kargakis commented:
@wanghaoran1988 I am ok with moving just the tests if we need to do more work in order to split the workload-specific utilities.

wanghaoran1988 commented:
@k8s-bot pull-kubernetes-federation-e2e-gce test this

wanghaoran1988 commented:
@k8s-bot pull-kubernetes-federation-e2e-gce test this

kargakis commented:
@wanghaoran1988 that job is broken and does not affect the queue I think.

wanghaoran1988 commented:
@kargakis Thanks :)

enisoc commented:
/approve no-issue

wanghaoran1988 commented:
/retest

ixdy commented:
/approve

wanghaoran1988 commented:
/retest

kargakis commented on test/e2e/workload/OWNERS:
```diff
@@ -0,0 +1,10 @@
+- kargakis
+- mfojtik
+reviewers:
+- janetkuo
```
Add sig-apps-reviewers

wanghaoran1988 replied:
@kargakis Should delete the other people, and only "sig-apps-reviewers" is enough ?

kargakis replied:
Yes

wanghaoran1988 replied:
@kargakis Updated

kargakis commented on test/e2e/kubectl.go:
```diff
@@ -61,6 +61,7 @@ import (
 	"k8s.io/kubernetes/test/e2e/framework"
 	"k8s.io/kubernetes/test/e2e/generated"
 	"k8s.io/kubernetes/test/e2e/scheduling"
+	. "k8s.io/kubernetes/test/e2e/workload"
```
Why do we need this import here?

kargakis replied:
At the very lease, don't import with . so we can know where is the package used.

wanghaoran1988 replied:
@kargakis Done

kargakis commented on test/e2e/framework/rs_util.go:
```diff
@@ -0,0 +1,194 @@
+		}
+		return err
+	}
+	reaper, err := kubectl.ReaperFor(extensionsinternal.Kind("ReplicaSet"), internalClientset)
```
Open a separate issue to remove the cli reaper from here and instead do a server-side deletion

wanghaoran1988 replied:
opened here:https://github.com/kubernetes/kubernetes/issues/47832

kargakis commented on test/e2e/kubectl.go:
```diff
@@ -205,7 +207,7 @@ var _ = framework.KubeDescribe("Kubectl alpha client", func() {
 
 		It("should create a ScheduledJob", func() {
-			framework.SkipIfMissingResource(f.ClientPool, ScheduledJobGroupVersionResource, f.Namespace.Name)
+			framework.SkipIfMissingResource(f.ClientPool, workload.ScheduledJobGroupVersionResource, f.Namespace.Name)
```
Instead of importing the whole package just for these two variables, copy them here.

wanghaoran1988 replied:
hmm, sounds reasonable.

wanghaoran1988 commented:
@kargakis Hi, all comments are addressed, would you mind add lgtm label, as every time I do a rebase, it will cost a lot time.

kargakis commented:
/lgtm

Note that this won't get merged before the code freeze is lifted

kargakis commented:
@wanghaoran1988 code freeze is lifted, we can merge this as soon as you rebase one more time :)

wanghaoran1988 commented:
@kargakis Cool, updated

kargakis commented:
/lgtm

kargakis commented:
/retest

wanghaoran1988 commented:
@kargakis Hoo, all tests passed now

kargakis commented:
/lgtm

kargakis commented:
/retest

kargakis commented:
Hooray! Thanks a lot @wanghaoran1988 !
